### PR TITLE
Add return types, round 1

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

In order to add return types to the codebase, v6, I think we should rely on the `.github/patch-types.php` script, which itself relies on `DebugClassLoader`. The reasoning is that if `DebugClassLoader` is able to parse `@return` annotations and turn them into real return types, it will also be able to parse them to trigger deprecations.

This is how I generated the attached patch, commit 1:

```sh
composer install --optimize-autoloader
SYMFONY_PATCH_TYPE_DECLARATIONS=force=1 php .github/patch-types.php
```

As you'll see, this leaves many `@return` annotations, and many methods are left with no return type at all.

We should solve this iteratively by:
1. adding `@return` annotations in lower branches where they miss
2. improve `DebugClassLoader` to make it able to parse more annotations (eg union types are not handled right now AFAIK)
3. run `.github/patch-types.php` again after merging up

The 2nd commit is manual edits.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
Green (failures are false positives.)

File: src/Symfony/Component/Lock/Store/RedisStore.php [hunk 1]
```diff
@@ -244,12 +244,7 @@ public function exists(Key $key)
         return (bool) $this->evaluate($script, (string) $key, [microtime(true), $this->getUniqueToken($key)]);
     }
 
-    /**
-     * Evaluates a script in the corresponding redis client.
-     *
-     * @return mixed
-     */
-    private function evaluate(string $script, string $resource, array $args)
+    private function evaluate(string $script, string $resource, array $args): mixed
```
Comment (Tobion) [inline src/Symfony/Component/Lock/Store/RedisStore.php:247]: function is missing a return statement at the end as psalm says

Comment (Tobion) [reply src/Symfony/Component/Lock/Store/RedisStore.php:247]: also there is a double return in line 272/274

Comment (nicolas-grekas) [reply src/Symfony/Component/Lock/Store/RedisStore.php:247]: good catch, fixed in https://github.com/symfony/symfony/pull/42065

File: src/Symfony/Contracts/Service/ServiceLocatorTrait.php [hunk 2]
```diff
@@ -50,10 +50,8 @@ public function has(string $id)
 
     /**
      * {@inheritdoc}
-     *
-     * @return mixed
      */
-    public function get(string $id)
+    public function get(string $id): mixed
```
Comment (Tobion) [inline src/Symfony/Contracts/Service/ServiceLocatorTrait.php:54]: `has` could also add a return type can't it?

Comment (nicolas-grekas) [reply src/Symfony/Contracts/Service/ServiceLocatorTrait.php:54]: In the end yes, but as I explained in the PR description, DebugClassLoader should do it, so we now need to figure out why it didn't add it on `has()`.

**Review by Tobion — APPROVED** [review]

File: src/Symfony/Bundle/FrameworkBundle/Routing/Router.php [hunk 3]
```diff
@@ -146,10 +146,8 @@ private function resolveParameters(RouteCollection $collection)
     /**
      * Recursively replaces placeholders with the service container parameters.
      *
-     * @param mixed $value The source which might contain "%placeholders%"
-     *
-     * @return mixed The source with the placeholders replaced by the container
-     *               parameters. Arrays are resolved recursively.
+     * @param mixed $value The source which might contain "%placeholders%".
+     *                     Arrays are resolved recursively.
```
Comment (stof) [inline src/Symfony/Bundle/FrameworkBundle/Routing/Router.php]: I'd rather move that sentence to the phpdoc description.

**Review by derrabus — APPROVED** [review]

**Pull request merged**